From 39d2882058345b5994680b8731848a0343878019 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@gmail.com>
Date: Sat, 4 Feb 2017 12:58:50 +0100
Subject: [PATCH 7/8] MIPS: BCM63XX: do not register SPI controllers

We now register them through DT, so no need to keep them here.

Signed-off-by: Jonas Gorski <jonas.gorski@gmail.com>
---
 arch/mips/bcm63xx/boards/board_bcm963xx.c | 7 -------
 1 file changed, 7 deletions(-)

--- a/arch/mips/bcm63xx/boards/board_common.c
+++ b/arch/mips/bcm63xx/boards/board_common.c
@@ -27,9 +27,7 @@
 #include <bcm63xx_dev_pci.h>
 #include <bcm63xx_dev_enet.h>
 #include <bcm63xx_dev_flash.h>
-#include <bcm63xx_dev_hsspi.h>
 #include <bcm63xx_dev_pcmcia.h>
-#include <bcm63xx_dev_spi.h>
 #include <bcm63xx_dev_usb_ehci.h>
 #include <bcm63xx_dev_usb_ohci.h>
 #include <bcm63xx_dev_usb_usbd.h>
@@ -236,10 +234,6 @@ int __init board_register_devices(void)
 	     bcm63xx_register_fallback_sprom(&board.fallback_sprom)))
 		pr_err(PFX "failed to register fallback SPROM\n");
 
-	bcm63xx_spi_register();
-
-	bcm63xx_hsspi_register();
-
 	bcm63xx_flash_register();
 
 	bcm63xx_led_data.num_leds = ARRAY_SIZE(board.leds);
--- a/arch/mips/bcm63xx/Makefile
+++ b/arch/mips/bcm63xx/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-2.0
 obj-y		+= clk.o cpu.o cs.o gpio.o irq.o nvram.o prom.o reset.o \
 		   setup.o timer.o dev-enet.o dev-flash.o dev-pcmcia.o \
-		   dev-rng.o dev-spi.o dev-hsspi.o dev-uart.o dev-wdt.o \
+		   dev-rng.o dev-uart.o dev-wdt.o \
 		   dev-usb-ehci.o dev-usb-ohci.o dev-usb-usbd.o usb-common.o \
 		   sprom.o
 obj-$(CONFIG_EARLY_PRINTK)	+= early_printk.o
--- a/arch/mips/bcm63xx/dev-hsspi.c
+++ /dev/null
@@ -1,48 +0,0 @@
-/*
- * This file is subject to the terms and conditions of the GNU General Public
- * License.  See the file "COPYING" in the main directory of this archive
- * for more details.
- *
- * Copyright (C) 2012 Jonas Gorski <jonas.gorski@gmail.com>
- */
-
-#include <linux/init.h>
-#include <linux/kernel.h>
-#include <linux/platform_device.h>
-
-#include <bcm63xx_cpu.h>
-#include <bcm63xx_dev_hsspi.h>
-#include <bcm63xx_regs.h>
-
-static struct resource spi_resources[] = {
-	{
-		.start		= -1, /* filled at runtime */
-		.end		= -1, /* filled at runtime */
-		.flags		= IORESOURCE_MEM,
-	},
-	{
-		.start		= -1, /* filled at runtime */
-		.flags		= IORESOURCE_IRQ,
-	},
-};
-
-static struct platform_device bcm63xx_hsspi_device = {
-	.name		= "bcm63xx-hsspi",
-	.id		= 0,
-	.num_resources	= ARRAY_SIZE(spi_resources),
-	.resource	= spi_resources,
-};
-
-int __init bcm63xx_hsspi_register(void)
-{
-	if (!BCMCPU_IS_6318() && !BCMCPU_IS_6328() && !BCMCPU_IS_6362() &&
-		!BCMCPU_IS_63268())
-		return -ENODEV;
-
-	spi_resources[0].start = bcm63xx_regset_address(RSET_HSSPI);
-	spi_resources[0].end = spi_resources[0].start;
-	spi_resources[0].end += RSET_HSSPI_SIZE - 1;
-	spi_resources[1].start = bcm63xx_get_irq_number(IRQ_HSSPI);
-
-	return platform_device_register(&bcm63xx_hsspi_device);
-}
--- a/arch/mips/bcm63xx/dev-spi.c
+++ /dev/null
@@ -1,60 +0,0 @@
-/*
- * This file is subject to the terms and conditions of the GNU General Public
- * License.  See the file "COPYING" in the main directory of this archive
- * for more details.
- *
- * Copyright (C) 2009-2011 Florian Fainelli <florian@openwrt.org>
- * Copyright (C) 2010 Tanguy Bouzeloc <tanguy.bouzeloc@efixo.com>
- */
-
-#include <linux/init.h>
-#include <linux/kernel.h>
-#include <linux/export.h>
-#include <linux/platform_device.h>
-#include <linux/err.h>
-#include <linux/clk.h>
-
-#include <bcm63xx_cpu.h>
-#include <bcm63xx_dev_spi.h>
-#include <bcm63xx_regs.h>
-
-static struct resource spi_resources[] = {
-	{
-		.start		= -1, /* filled at runtime */
-		.end		= -1, /* filled at runtime */
-		.flags		= IORESOURCE_MEM,
-	},
-	{
-		.start		= -1, /* filled at runtime */
-		.flags		= IORESOURCE_IRQ,
-	},
-};
-
-static struct platform_device bcm63xx_spi_device = {
-	.id		= -1,
-	.num_resources	= ARRAY_SIZE(spi_resources),
-	.resource	= spi_resources,
-};
-
-int __init bcm63xx_spi_register(void)
-{
-	if (BCMCPU_IS_6318() || BCMCPU_IS_6328() || BCMCPU_IS_6345())
-		return -ENODEV;
-
-	spi_resources[0].start = bcm63xx_regset_address(RSET_SPI);
-	spi_resources[0].end = spi_resources[0].start;
-	spi_resources[1].start = bcm63xx_get_irq_number(IRQ_SPI);
-
-	if (BCMCPU_IS_6338() || BCMCPU_IS_6348()) {
-		bcm63xx_spi_device.name = "bcm6348-spi",
-		spi_resources[0].end += BCM_6348_RSET_SPI_SIZE - 1;
-	}
-
-	if (BCMCPU_IS_3368() || BCMCPU_IS_6358() || BCMCPU_IS_6362() ||
-		BCMCPU_IS_6368() || BCMCPU_IS_63268()) {
-		bcm63xx_spi_device.name = "bcm6358-spi",
-		spi_resources[0].end += BCM_6358_RSET_SPI_SIZE - 1;
-	}
-
-	return platform_device_register(&bcm63xx_spi_device);
-}
--- a/arch/mips/include/asm/mach-bcm63xx/bcm63xx_dev_hsspi.h
+++ /dev/null
@@ -1,9 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef BCM63XX_DEV_HSSPI_H
-#define BCM63XX_DEV_HSSPI_H
-
-#include <linux/types.h>
-
-int bcm63xx_hsspi_register(void);
-
-#endif /* BCM63XX_DEV_HSSPI_H */
